src/mono/mono.proj

```diff
@@ -554,18 +560,19 @@
       <MonoToolchainPrebuiltOS Condition="$([MSBuild]::IsOSPlatform('OSX'))">darwin-x86_64</MonoToolchainPrebuiltOS>
       <MonoToolchainPrebuiltOS Condition="'$(HostOS)' == 'windows'">windows-x86_64</MonoToolchainPrebuiltOS>
       <_MonoRuntimeFilePath>$(MonoObjDir)out\lib\$(MonoFileName)</_MonoRuntimeFilePath>
-      <_LinuxAbi Condition="'$(TargetsAndroid)' != 'true' and '$(TargetsLinuxBionic)' != 'true'">gnu</_LinuxAbi>
-      <_LinuxAbi Condition="'$(TargetsAndroid)' == 'true' or '$(TargetsLinuxBionic)' == 'true'">android</_LinuxAbi>
+      <_LinuxAbi Condition="'$(TargetsAndroid)' != 'true' and '$(TargetsLinuxBionic)' != 'true' and '$(TargetsLinuxMusl)' != 'true'">linux-gnu</_LinuxAbi>
+      <_LinuxAbi Condition="'$(TargetsAndroid)' != 'true' and '$(TargetsLinuxBionic)' != 'true' and '$(TargetsLinuxMusl)' == 'true'">alpine-linux-musl</_LinuxAbi>
```

Comment:
Is there a `{arch}-linux-musl-{tool}` variant available (without the `alpine-` part), which would work for AlpineLine and VoidLinuxMusl alike?

Reply:
Hmm good question. I don't know how void does their crosscompilation, as really this code is only relevant then.

Reply:
I use the portable linux-x64 and linux-musl-x64 installer binaries on Void Linux amd64 and musl amd64 respectively (since https://github.com/dotnet/runtime/issues/28265), it works well. I haven't cross-compiled mono for Void. :)

Judging by the `system types` at https://gcc.gnu.org/install/configure.html, distro can choose to use `linux-musl` target.

Reply:
Thus if TargetsLinuxMusl _LinuxAbi should be linux-musl. Thus perhaps I should add in a TargetsLinuxAlpine check.

Reply:
Indeed, Alpine's crosscompilation framework uses `{arch}-alpine-linux-musl-{tool}`. If we were to want to target `void-{armv6,s390x}`, then I'd think that having seperate checks for `alpine`and `linux-musl` would be relevant. Given the former being unlikely anytime soon, and judging by having already included alpine-specific code in `toolchain.cmake`, I'd consider it out-of-scope of this PR.

Reply:
I think we should be forward looking and add an explicit `TargetsAlpineLinux` property and check (it's an easy addition to the root Directory.Build.props).